### PR TITLE
feat(loader): Add new dynamic cdn options to js loader

### LINKED ISSUE #44225: Add dynamic sdk options to JavaScript loader backend
Playground PR: https://github.com/getsentry/sentry/pull/44237

```[tasklist]
### Tasks
- [ ] https://github.com/getsentry/sentry/pull/44228
- [ ] https://github.com/getsentry/sentry/pull/44346
- [ ] https://github.com/getsentry/sentry/pull/44492
- [ ] https://github.com/getsentry/sentry/pull/44496
- [ ] https://github.com/getsentry/sentry/pull/44876
```

### PR DESCRIPTION
**Pull request by AbhiPrasad**
resolves https://github.com/getsentry/sentry/issues/44225

Building off of https://github.com/getsentry/sentry/pull/44496, we add the new project settings to the js sdk loader. Currently this should have 0 impact on any existing users, as the options under `DynamicSdkLoaderOption` should never be set for them.

As per @mitsuhiko's recommendation, we remove the new dynamic loader endpoint in https://github.com/getsentry/sentry/pull/44346, in favor of putting all the functionality in the original loader.

### CODE REVIEW

File: tests/sentry/web/frontend/test_js_sdk_loader.py [hunk 1]
```diff
@@ -104,6 +109,86 @@ def test_greater_than_v7_returns_es5(
...
 
+    @mock.patch("sentry.loader.browsersdkversion.load_version_from_file", return_value=["7.37.0"])
+    @mock.patch(
+        "sentry.loader.browsersdkversion.get_selected_browser_sdk_version", return_value="latest"
+    )
+    def test_bundle_kind_modifiers(self, load_version_from_file, get_selected_browser_sdk_version):
+        settings.JS_SDK_LOADER_DEFAULT_SDK_URL = "https://browser.sentry-cdn.com/%s/bundle%s.min.js"
+        settings.JS_SDK_LOADER_SDK_VERSION = "7.32.0"
+
+        for data, expected in [
```
Comment (AbhiPrasad) [inline tests/sentry/web/frontend/test_js_sdk_loader.py:120]: couldn't make `@mock.patch` and `pytest.mark.parametrize` play nicely with each other, so elected for the manual for loop.

File: src/sentry/web/frontend/js_sdk_loader.py [hunk 2]
```diff
@@ -25,18 +38,57 @@ class JavaScriptSdkLoader(BaseView):
...
+        if has_replay:
+            bundle_kind_modifier += ".replay"
+            isLazy = False
+
+        # From JavaScript SDK version 7 onwards, the default bundle code is ES6, however, in the loader we
+        # want to provide the ES5 version. This is why we need to modify the requested bundle name here.
+        #
+        # If we are loading replay, do not add the es5 modifier, as those bundles are
+        # ES6 only.
+        if sdk_version >= Version("7.0.0") and not has_replay:
```
Comment (mydea) [inline src/sentry/web/frontend/js_sdk_loader.py:68]: Hmm, just thinking, do we need to handle this somehow, as replay generally does not work before 7?

File: tests/sentry/web/frontend/test_js_sdk_loader.py [hunk 3]
```diff
@@ -6,10 +6,15 @@
...
 from django.urls import reverse
 
+from sentry.loader.dynamic_sdk_options import DynamicSdkLoaderOption
 from sentry.testutils import TestCase
 
 
 class JavaScriptSdkLoaderTest(TestCase):
+    # def setUp(self):
+    # self.project = self.create_project()
+    # self.project_key = self.create_project_key(project=self.project)
```
Comment (Lms24) [inline tests/sentry/web/frontend/test_js_sdk_loader.py:16]: l: Do we still need these comments? If not let's remove them

File: src/sentry/templates/sentry/js-sdk-loader.min.js.tmpl [hunk 4]
```diff
@@ -1,4 +1,9 @@
...
+!1,p=[],m=function(a){("e"in a||"p"in a||a.f&&-1<a.f.indexOf("capture")||a.f&&-1<a.f.indexOf("showReportDialog"))&&n&&t(p);m.data.push(a)};m.data=[];c[k]=c[k]||{};c[k].onLoad=function(a){p.push(a);n&&!B||t(p)};c[k].forceLoad=function(){B=!0;n&&setTimeout(function(){t(p)})};"init addBreadcrumb captureMessage captureException captureEvent configureScope withScope showReportDialog".split(" ").forEach(function(a){c[k][a]=function(){m({f:a,a:arguments})}});var u=c[q];c[q]=function(a,f,d,b,g){m({e:[].slice.call(arguments)});
+u&&u.apply(c,arguments)};var v=c[r];c[r]=function(a){m({p:"reason"in a?a.reason:"detail"in a&&"reason"in a.detail?a.detail.reason:a});v&&v.apply(c,arguments)};n||setTimeout(function(){t(p)})})(window,document,"script","onerror","onunhandledrejection","Sentry","{{ publicKey|safe }}","{{ jsSdkUrl|safe }}",{{ config|to_json|safe }},{{ isLazy|safe }});
```
Comment (Lms24) [inline src/sentry/templates/sentry/js-sdk-loader.min.js.tmpl:4]: m: Not sure if we're missing something but do we actually need these lines? Seems like we have the two versions of the minified loader template in this file and we should have only one, right?

**Review by Lms24 — APPROVED** [review]
Reviewed with @lforst - generally looks good to us - one question remains.

File: tests/sentry/web/frontend/test_js_sdk_loader.py (hunk 3 above)
Comment (lforst) [reply tests/sentry/web/frontend/test_js_sdk_loader.py]: [fb27bde](https://github.com/getsentry/sentry/pull/44876/commits/fb27bde54a9707b228b9d0aff259195d2e02ebbc)

File: src/sentry/templates/sentry/js-sdk-loader.min.js.tmpl (hunk 4 above)
Comment (lforst) [reply src/sentry/templates/sentry/js-sdk-loader.min.js.tmpl]: [18a2502](https://github.com/getsentry/sentry/pull/44876/commits/18a250229cc9d05887c95e8ab0ca7393a6c1b4b6)

**Review by Lms24 — APPROVED** [review]

**Review by dashed — APPROVED** [review]

File: tests/sentry/web/frontend/test_js_sdk_loader.py (hunk 1 above)
Comment (markstory) [reply tests/sentry/web/frontend/test_js_sdk_loader.py:116]: Yeah pytest.parametrize and unittest.TestCase don't play nice either.

**Review by markstory — COMMENTED** [review]
Changes look good to me from an API conventions point of view.

**Comment by lforst** [comment]
@markstory Thanks for taking a look :)

**Pull request merged**